### ISSUE TITLE
Remote UI client blocks when ATTENTION dialog is displayed during VimEnter

### ISSUE THREAD

**Issue by tweekmonster** [neovim/neovim#5739]
- `nvim --version`: NVIM v0.2.0-107-gd2bb979
- Vim (version: ) behaves differently? N/A
- Operating system/version: Ubuntu 16.04.1 LTS
- Terminal name/version: N/A
- `$TERM`: N/A

I have an `autocmd` that loads sessions on `VimEnter`.  If a file that's being loaded has an existing swap file, the `ATTENTION` dialog is displayed.

When running as a remote UI server, the dialog screen is not sent to the client and input from the client is blocked.  I can't come up with a simple minimal test for this, but this is an easy way to get a dangling swap file:

1. `nvim -u NORC +'autocmd VimEnter * edit ~/example.txt`
2. Press ctrl+z to suspend
3. `kill -KILL %1` to kill suspended process
4. Repeat step 1 to get the prompt

If the remote UI server is setup to run with the autocmd above and `~/example.txt` has an existing swap file, the remote UI client will block.

I've concluded that blocking only happens with the `ATTENTION` dialog.  `input()` doesn't block on the client:

```
nvim -u NORC +'autocmd VimEnter * echo input("hello\nthere")'
```

But, only the last line is displayed.  Normally it will send the first line, scroll the screen up, then display the second line.

The earliest point I found for this problem is in [main.c#L515](https://github.com/neovim/neovim/blob/ddfac951eab5d392eb0bc36748bc3d586203cc11/src/nvim/main.c#L515), but I couldn't get much further than that.  I think that the issue is caused because `findswapname()` runs before the main loop starts.

Ping @garyburd to let him know that go-client isn't causing this issue AFAIK.

**Comment by justinmk** [comment]
How about we re-send the screen state whenever a UI attaches?

**Comment by tweekmonster** [comment]
I assumed that's how it already worked.  Maybe I just didn't notice because the app I'm working on intrinsically sends `try_resize` after attaching.

I'm not sure it's currently able to anyways.  I added `ILOG("attached");` to `nvim_ui_attach()` and it's never emitted in the case of the `ATTENTION` dialog.  I guess in the case of using `input()` try resize doesn't succeed so the screen doesn't update.

I noticed something else strange about the remote UI but it should probably be another issue: if a Terminal window is present, resizing the terminal window causes a full screen redraw.
